walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,101 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 public final class FindMeetingQuery {
+  /**
+   * Method that finds the available meeting times given events for the days, and request
+   * for the new meeting.
+   *
+   * @param events: Collection of events already scheduled in the day
+   * @param request: Details of the new meeting which we try to find time for
+   * @return list of potential TimeRanges to schedule the new event
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Return empty list if duration exceeds length of day.
+    if (request.getDuration() >= TimeRange.END_OF_DAY) {
+      return new ArrayList<>();
+    }
+
+    ArrayList<String> attendees = new ArrayList<String>(request.getAttendees());
+    ArrayList<String> optionalAttendees = new ArrayList<String>(request.getOptionalAttendees());
+    ArrayList<TimeRange> meetingTimes = new ArrayList<TimeRange>();
+    ArrayList<TimeRange> meetingTimesWithOptionals = new ArrayList<TimeRange>();
+    ArrayList<TimeRange> requiredSlots = new ArrayList<TimeRange>();
+    ArrayList<TimeRange> withOptionalSlots = new ArrayList<TimeRange>();
```

Comment:
Use ImmutableList if list isn't changed outside of the method
https://www.baeldung.com/java-immutable-list